src/main/java/org/bytedeco/javacpp/Loader.java

```diff
@@ -765,15 +768,16 @@ public static File extractResource(URL resourceURL, File directoryOrFile,
                         File file = new File(directoryOrFile, entryName.substring(jarEntryName.length()));
                         if (entry.isDirectory()) {
                             file.mkdirs();
+                            file.setLastModified(entryTimestamp);
                         } else if (!cacheDirectory || !file.exists() || file.length() != entrySize
-                                || file.lastModified() != entryTimestamp || !file.equals(file.getCanonicalFile())) {
```

Comment:
I'm not sure what the intention of the comparison with the canonical file is, but the file is not canonicalized in the called extractResource() method. Furthermore canonicalization does not (or at least should not) change the effective target the file is pointing to, so the metadata checked before should not change because of that.

Reply:
Symbolic links get created and we need to resolve them to prevent creating cycles.

Reply:
Can you please explain how this prevents cycles or is there an example anywhere? Because I don't understand it.
I'm asking because avoiding the canonicalization had a significant impact on the runtime improvement (around one third of the runtime reduction).
So if we can avoid this call or at least have a cheaper pre-check (e.g. `canCreateSymbolicLink==true`? or `Files.isSymbolicLink()==true`, I'm not sure if this is much faster), it should make it faster.
But of course the logic has to be correct in any case.

Reply:
There can be symbolic links anywhere in the path, not just the last file in the path.

I'm pretty sure we can disable those checks when org.bytedeco.javacpp.cachedir.nosubdir is set though, and I think that's what you're already using, right? If so, I'd be happy to skip all that when that flag is set. How does that sound?